File: 1310.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n,c;
    
    while(cin >> n >> c)
    {
        vector<int> dias;
        int dia,maior=0;
        
        dias.push_back(0);
        
        for(int i=0;i<n;i++)
        {
            cin >> dia;
            dias.push_back(dias[i]+dia);
        }
        
        for(int i=1;i<=n;i++)
        {
            for(int j=0;j<i;j++)
            {
                int lucro = dias[i] - dias[j] - (i-j)*c;
                maior = max(maior,lucro);
            }
        }
        
        cout << maior << endl;
        
    }
    return 0;

File: 1310.py
while True:
  try:
    maior = 0
    dia = 0
    N = int(input())
    custoPorDia = int(input())
    receita = []
    receita.append(0)
    for i in range(1,N+1):
      receita.append(receita[i-1]+int(input())-custoPorDia)
    for i in range(1,N+1):
      for j in range(i):
        lucro = receita[i]-receita[j]
        maior = max(maior,lucro)
    print(maior)
    
  except EOFError:
    break
